src/main/scala/org/scalameter/Executor.scala

```diff
@@ -1,18 +1,21 @@
 package org.scalameter
 
 
-import collection._
-import compat._
-import utils.{withGCNotification, Tree}
+import utils.Tree
 
 
 
 trait Executor {
 
+  def measurer: Measurer
+
   def run[T](setuptree: Tree[Setup[T]], reporter: Reporter, persistor: Persistor): Tree[CurveData] = {
+    val context = measurer.prepareContext(setuptree.context)
```

Comment:
Careful - there is a Context on the setup class, but also a context on
the Tree itself. Mapping would only modify the context in Setup.

On 7/9/2015 2:32 AM, Krzysztof Janosz wrote:
>>    def run[T](setuptree: Tree[Setup[T]], reporter: Reporter, persistor: Persistor): Tree[CurveData] = {
>> +    val context = measurer.prepareContext(setuptree.context)
> 
> Btw, since I should also adjust context in all setups is it OK to just call the `map` method on tree and substitute each setup context with result of the `initialConfig`?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/scalameter/scalameter/pull/135/files#r34213878
> 

-- 
Aleksandar Prokopec